lib/matplotlib/tests/test_gridspec.py

```diff
@@ -48,3 +76,8 @@ def test_subplotspec_args():
         gs = gridspec.GridSpecFromSubplotSpec(2, 1, subplot_spec=axs[0])
     with pytest.raises(TypeError, match="subplot_spec must be type SubplotSpec"):
         gs = gridspec.GridSpecFromSubplotSpec(2, 1, subplot_spec=axs)
+
+    s = gridspec.SubplotParams(.1, .1, .9, .9)
+    p = unittest.mock.MagicMock()
+    s._repr_pretty_(p, False)
+    assert 'SubplotParams' in p.method_calls[0].args[0]
```

Comment:
This shouldn't be in here. If you want to do the repr, implement `__repr__` and add a dediated test. But I'd rather move this to a separate PR as that's orthogonal to the fix here.